### DISCUSSION TITLE
Deciding on windRose and interpolant functions in long-term bin weighting

### DISCUSSION THREAD

**Discussion by paulf81** [NREL/flasc#145] [General]
@misi9170 , @Bartdoekemeijer  @ejsimley  @christiannvaughn 

Opening a discussion here to reconcile the requests in Issue #125  and #126 (either closing both in favor of a new one, or refining),

Current main idea is WindRose should be improved in FLORIS 4.0 and should provide capabilities to handle most of what is desired in FLASC 2.0

**Comment by Bartdoekemeijer** [comment]
These are my main use cases for a WindRose class:

1. Be able to ingest a wind direction vs. wind speed vs. frequency table directly.
2. Be able to ingest a wind direction and wind speed measurement timeseries and derive the frequencies internally.
3. Generate a linear interpolant / upsample to higher resolution wind direction and wind speed steps through linear interpolation.
4. Preferably also ingest a wind direction vs. wind speed vs. mean turbulence intensity table directly.
5. Preferably also ingest a wind direction, wind speed and turbulence intensity measurement timeseries and derive the mean turbulence intensity per wind direction/speed bin internally.
6. Produce a wind rose plot.

Happy to brainstorm on how this all best pieces together.

**Reply by paulf81** [reply]
I think we've been coming towards a conclusion that WindRose will take on more fundamental responsibilities in 4.0, basically enabling all ws/wd matrix type runs in a convenient way.  So I would say yes to all on your list, plus:

  7. Be able to be supplied directly to reinitialize to drive FLORIS off of a wind rose (versus series of values)
  8. Be able to build tables off of relationships (TI based on wind speed, market price based on wind speed) etc
  9.
